### PR TITLE
API/UCT: added UCT_IFACE_FLAG_EP_KEEPALIVE caps

### PR DESCRIPTION
**Pull request by hoopoepg**
## What

- added iface capability UCT_IFACE_FLAG_EP_KEEPALIVE

## Why ?

- to  identify EP's which can provide keepalive functionality
  without UCX involving

## How ?

- API changed

### CODE REVIEW

**Review by brminich — APPROVED** [review]

**Comment by hoopoepg** [comment]
@tonycurtis @shamisp any comments?

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -412,6 +412,9 @@ typedef enum uct_atomic_op {
                                                        and it may also be invoked when uct_worker_progress()
                                                        is called. */
 
+        /* Keepalive */
+#define UCT_IFACE_FLAG_EP_KEEPALIVE   UCS_BIT(46) /**< EP has built-in keepalive feature */
```
Comment (yosefe) [inline src/uct/api/uct.h:416]: /**< Transport endpoint has built-in keepalive feature, which guarantees the error callback on the transport interface will be called if the communication channel with remote peer is broken, even if there are no outstanding send operations */

Comment (hoopoepg) [reply src/uct/api/uct.h]: updated

**Review by tonycurtis — APPROVED** [review]

**Comment by shamisp** [comment]
👍

**Review by dmitrygx — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Comment by hoopoepg** [comment]
@yosefe ok to approve?

**Review by yosefe — APPROVED** [review]

**Pull request merged**